vms/platformvm/txs/fee/calculator_test.go

```diff
@@ -0,0 +1,249 @@
+		},
+		{
+			name:       "TransferSubnetOwnershipTx pre EUpgrade",
+			chainTime:  upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: transferSubnetOwnershipTx,
+			expected:   feeTestsDefaultCfg.TxFee,
+		},
+		{
+			name:      "AddPermissionlessValidatorTx Primary Network pre EUpgrade",
+			chainTime: upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: func() txs.UnsignedTx {
+				return addPermissionlessValidatorTx(constants.PrimaryNetworkID)
+			},
+			expected: feeTestsDefaultCfg.AddPrimaryNetworkValidatorFee,
+		},
+		{
+			name:      "AddPermissionlessValidatorTx Subnet pre EUpgrade",
+			chainTime: upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: func() txs.UnsignedTx {
+				subnetID := ids.GenerateTestID()
+				require.NotEqual(t, constants.PrimaryNetworkID, subnetID)
+				return addPermissionlessValidatorTx(subnetID)
+			},
+			expected: feeTestsDefaultCfg.AddSubnetValidatorFee,
+		},
+		{
+			name:      "AddPermissionlessDelegatorTx Primary Network pre EUpgrade",
+			chainTime: upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: func() txs.UnsignedTx {
+				return addPermissionlessDelegatorTx(constants.PrimaryNetworkID)
+			},
+			expected: feeTestsDefaultCfg.AddPrimaryNetworkDelegatorFee,
+		},
+		{
+			name:      "AddPermissionlessDelegatorTx pre EUpgrade",
+			chainTime: upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: func() txs.UnsignedTx {
+				subnetID := ids.GenerateTestID()
+				require.NotEqual(t, constants.PrimaryNetworkID, subnetID)
+				return addPermissionlessDelegatorTx(subnetID)
+			},
+			expected: feeTestsDefaultCfg.AddSubnetDelegatorFee,
+		},
+		{
+			name:       "BaseTx pre EUpgrade",
+			chainTime:  upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: baseTx,
+			expected:   feeTestsDefaultCfg.TxFee,
+		},
+		{
+			name:       "ImportTx pre EUpgrade",
+			chainTime:  upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: importTx,
+			expected:   feeTestsDefaultCfg.TxFee,
+		},
+		{
+			name:       "ExportTx pre EUpgrade",
+			chainTime:  upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: exportTx,
+			expected:   feeTestsDefaultCfg.TxFee,
+		},
+		{
+			name:      "RewardValidatorTx pre EUpgrade",
+			chainTime: upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: func() txs.UnsignedTx {
+				return &txs.RewardValidatorTx{
+					TxID: ids.GenerateTestID(),
+				}
+			},
+			expected: 0,
+		},
+		{
+			name:      "AdvanceTimeTx pre EUpgrade",
+			chainTime: upgrades.EUpgradeTime.Add(-1 * time.Second),
+			unsignedTx: func() txs.UnsignedTx {
+				return &txs.AdvanceTimeTx{
+					Time: uint64(time.Now().Unix()),
+				}
+			},
+			expected: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			uTx := tt.unsignedTx()
+			fc := NewStaticCalculator(feeTestsDefaultCfg, upgrades)
+			require.Equal(t, tt.expected, fc.CalculateFee(uTx, tt.chainTime))
+		})
+	}
+}
+
+func addValidatorTx() txs.UnsignedTx {
```

Comment:
This is forward looking. In next PR I will use non-empty txs since with dynamic fee, fees will depend on the transaction content. So keeping the function margially reduces diffs down the line